_posts/2016-08-26-babili.md

```diff
@@ -0,0 +1,183 @@
+
+In addition, there are also specific optimizations we can make when we minify ES2015+ code.
+
+## Usage
+
+### Babel Preset
+
+If you already use Babel, you can just add the [babili](https://github.com/babel/babili#babel-preset) preset (babel-preset-babili) to your config.
+
+You probably want to enable this only in production, so use the [env option](http://babeljs.io/docs/usage/babelrc/#env-option) which uses either `process.env.BABEL_ENV` or `process.env.NODE_ENV`
+
+```bash
+$ npm install babel-preset-babili --save-dev
+```
+
+```js
+// previous .babelrc
+{ "presets": ["es2015"] }
+// .babelrc
+{
+  "presets": ["es2015"],
+  "env": {
+    "production": {
+      "presets": ["babili"]
+    }
+  }
+}
+```
+
+### Babili CLI
+
+If you don't use Babel, you can use our standalone cli tool [`babili`](https://github.com/babel/babili#cli). (Currently it's just a wrapper for babel-cli + the preset). You could run this after transpiling (or not) in place of uglify.
+
+```bash
+$ npm install babili --save-dev
+```
+
+```bash
+# equivalent to 
+# babel src -d lib --presets=babili
+$ babili src -d lib
+```
+
+### Webpack
+
+You can either just use the preset option above with `babel-loader`, or use it seperately with the [babili-webpack-plugin](https://github.com/boopathi/babili-webpack-plugin) (made by [@boopathi](https://github.com/boopathi/), who also works on babili).
+
+```bash
+$ npm install babili-webpack-plugin --save-dev
+```
+
+```js
+// webpack.config.js
+const BabiliPlugin = require("babili-webpack-plugin");
+module.exports = {
+  entry: //...,
+  output: //...,
+  plugins: [
+    new BabiliPlugin(options)
+  ]
+}
+```
+
+We want to have a better story with integration with webpack/bundlers in the near future! Ref [#100](https://github.com/babel/babili/issues/100).
+
+## Pros/Cons
+
+Uglify Pros
+- No change to existing tooling if you are already minifying.
+- Battle-tested/production ready (been around for years, and has wide adoption so there's less bugs/issues).
+- It's fast and outputs small code already.
+
+Uglify Cons
+- Custom parser/tooling, so difficult to output/minify ES6 and more.
+- Not modular, no way to create own plugins/minification strategies.
+
+Babili Pros:
+- ES6+ aware (nothing special needs to be done because we can use the babylon parser) and Babel will update as standards/browsers update.
+- Uses the existing babel toolchain, can consume as a babel preset or standalone.
+- Potential for custom smart transforms for React/Flow, etc.
+- Could use flow/typescript annotations to help with minification.
```

Comment:
Does Babel now support TypeScript annotation parsing like it does Flow or is this referring to something else?